### PR TITLE
Implemented the assignment of AssociatedXXXGroup for CSOM and SSOM

### PR DESCRIPTION
**Pull request by andreasblueher**
Regarding my issue #1108 I've changed the corresponding handler, while the validators still remain empty.

### CODE REVIEW

File: SPMeta2/SPMeta2.CSOM/ModelHandlers/SecurityGroupModelHandler.cs [hunk 1]
```diff
@@ -245,6 +239,36 @@ private void DeploySiteGroup(object modelHost, SiteModelHost siteModelHost, Defi
...
+            if (securityGroupModel.IsAssociatedMemberGroup)
+                web.AssociatedMemberGroup = currentGroup;
+
+            if (securityGroupModel.IsAssociatedOwnerGroup)
+                web.AssociatedOwnerGroup = currentGroup;
+
+            if (securityGroupModel.IsAssociatedVisitorsGroup)
+                web.AssociatedVisitorGroup = currentGroup;
+
+            // Does this need to be called? Would it work using the securityGroupModel
```
Comment (andreasblueher) [inline SPMeta2/SPMeta2.CSOM/ModelHandlers/SecurityGroupModelHandler.cs:255]: Could you please tell me if this is needed or if I should remove the code before you merge the PR?

**Comment by andreasblueher** [comment]
I've looked into the validators for prevalidating definitions and I was wondering if there's a way to check if more than 1 securitygroupdefinition has "AssociatedOwnerGroup = yes" assigned to it, because that could cause issues.

**Comment by SubPointSupport** [comment]
Yes, it is possible. We have an infrastructure to check three types of validity:
* a single definition itself
* a group of definition of the same type - multiple definitions
* relationships between definitions

All is under this namespace:
> https://github.com/SubPointSolutions/spmeta2/tree/master/SPMeta2/SPMeta2/Validation/Validators

In this case, it would be a "collection" validator similar to FieldCollectionValidator
> https://github.com/SubPointSolutions/spmeta2/blob/master/SPMeta2/SPMeta2/Validation/Validators/Collections/FieldCollectionValidator.cs

This API is rather old. There is another validation API under the following namespace:
> https://github.com/SubPointSolutions/spmeta2/tree/master/SPMeta2/SPMeta2/Services/Impl/Validation

This one works similar to model handlers and has a broader access to the model tree.

**Comment by SubPointSupport** [comment]
Most likely this PR won't go. Waiting for @andreasblueher's review on the following work:
> https://github.com/SubPointSolutions/spmeta2/issues/1108#issuecomment-390963801

**Pull request closed without merging**